app/models/email.rb

```diff
@@ -0,0 +1,26 @@
+class Email < ApplicationRecord
+  belongs_to :notification
+  validates :subject, :body, :notification, presence: true
+
+  def self.create_from_params!(params)
+    build_from_params(params).tap(&:save!)
+  end
+
+  def self.build_from_params(params)
+    new.tap do |instance|
```

Comment:
Not sure on the benefit of using `tap` here.

Reply:
I just thought it was neater than doing: 

```
instance = new
instance.foo = params[:foo]
instance.bar = params[:bar]
#etc...
instance
```

Reply:
I suppose we could do `new(<the params>)`

Reply:
Yeah, I was thinking `new()` but I don't mind really, was just wondering if there was another reason.

Reply:
Oh actually we couldn't because we need to do some stuff to them first.

Reply:
Oh good point, yep.

Reply:
The params aren't actually 'the params' for the model